amp.php

```diff
@@ -159,10 +173,12 @@ function amp_prepare_render() {
  * @since 0.1
  */
 function amp_render() {
-
 	// Note that queried object is used instead of the ID so that the_preview for the queried post can apply.
-	amp_render_post( get_queried_object() );
-	exit;
+	$post = get_queried_object();
+	if ( $post instanceof WP_Post ) {
```

Comment:
Should it be exciting even if the page can't be rendered?

Reply:
Yes, it is very exciting. 😉 

In practice this function only ever gets called if the queried object is already guaranteed to be a post, so this is more for the sake of static analysis. But yeah, it could be better to not exit here.

Reply:
Haha, very exciting indeed 😂

Reply:
I think we can just leave this as-is right now. We'll need to revisit how this works when we start rendering non-posts in AMP anyway.